lib/pages/modlog_page.dart

```diff
@@ -0,0 +1,519 @@
+  final String instanceHost;
+  final String name;
+  final int communityId;
+
+  const ModlogPage.forInstance({
+    @required this.instanceHost,
+  })  : assert(instanceHost != null),
+        communityId = null,
+        name = instanceHost;
+
+  const ModlogPage.forCommunity({
+    @required this.instanceHost,
+    @required this.communityId,
+    @required String communityName,
+  })  : assert(instanceHost != null),
+        assert(communityId != null),
+        assert(communityName != null),
+        name = '!$communityName';
+
+  @override
+  Widget build(BuildContext context) {
+    final page = useState(1);
+    // will be set true when a fetch returns 0 results
+    final isDone = useState(false);
+
+    final modlogFut = useMemoized(
+      () => LemmyApiV2(instanceHost).run(
+        GetModlog(
+          communityId: communityId,
+          page: page.value,
+        ),
+      ),
+      [page.value],
+    );
+
+    return Scaffold(
+      appBar: AppBar(title: Text("$name's modlog")),
+      body: SingleChildScrollView(
+        child: Column(
+          children: [
+            FutureBuilder<Modlog>(
+              key: ValueKey(modlogFut),
+              future: modlogFut,
+              builder: (context, snapshot) {
+                if (!snapshot.hasData) {
+                  return const Center(child: CircularProgressIndicator());
+                } else if (snapshot.hasError) {
+                  return Center(
+                      child: Text('Error: ${snapshot.error?.toString()}'));
+                }
+                final modlog = snapshot.data;
+
+                if (modlog.added.length +
+                        modlog.addedToCommunity.length +
+                        modlog.banned.length +
+                        modlog.bannedFromCommunity.length +
+                        modlog.lockedPosts.length +
+                        modlog.removedComments.length +
+                        modlog.removedCommunities.length +
+                        modlog.removedPosts.length +
+                        modlog.stickiedPosts.length ==
+                    0) {
+                  WidgetsBinding.instance
+                      .addPostFrameCallback((_) => isDone.value = true);
+
+                  return const Center(child: Text('no more logs to show'));
+                }
+
+                return _ModlogTable(modlog: modlog);
+              },
+            ),
+            Padding(
+              padding: const EdgeInsets.all(8),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  TextButton(
+                    onPressed: page.value != 1 ? () => page.value-- : null,
+                    child: const Icon(Icons.skip_previous),
+                  ),
+                  TextButton(
+                    onPressed: isDone.value ? null : () => page.value++,
+                    child: const Icon(Icons.skip_next),
+                  ),
```

Comment:
hide the buttons when the spinner is present

Reply:
I disagree, what if someone wants to quickly skip a few pages?

Reply:
in that case maybe change it so that if theres no table then buttons are fixed to the bottom of the page. cuz it kinda looks ugly with those buttons in the middle

Reply:
Fixed footer... I am getting html/css flashbacks